lib/snapshot/reader.js

```diff
@@ -55,7 +55,7 @@ async function loadFromUpstream() {
       throw new Error('failed retrieving latest snapshot');
     }
 
-    lastModified = new Date(response.headers['Last-Modified']); // TODO caps?
+    lastModified = new Date(Date.parse(response.headers['last-modified']));
```

Comment:
What's the `Date.parse` doing here? Pretty sure node's `Date` constructor accepts anything a real browser would send? (Relevant as there's identical code in Homebass.)

Reply:
you're right.
I had thought `Date`'s constructor couldn't handle it, but it was some other bug I had that read the `last modified` as `undefined` but I thought it just can't handle strings.
removed!
that said, though, what do you think the agent should do if Homebase replies with a new snapshot but doesn't attach the `last-modified` header?

Reply:
The most important failure here is never getting new data, not overloading homebase (as we're generating one request every N hours); so we should just re-request without `if-modified-since` in that case?

Reply:
you mean re-request immediately, or on the next iteration?
because if we got a response with a snapshot, just without the `last-modified` header, then allegedly we got the latest snapshot (according to what Homebase thinks), we just don't know when it's from for the next request.